builtin/logical/database/path_roles.go

```diff
@@ -578,6 +594,18 @@ func (b *databaseBackend) pathStaticRoleCreateUpdate(ctx context.Context, req *l
 			return logical.ErrorResponse("could not parse rotation_schedule"), nil
 		}
 		role.StaticAccount.Schedule = *sched
+	} else if rotationWindowOk {
+		return logical.ErrorResponse("rotation_window is invalid without use of rotation_schedule"), nil
+	}
+
+	if rotationWindowOk {
+		rotationWindowSeconds := rotationWindowSecondsRaw.(int)
```

Comment:
The decisions might have changed so apologies if I'm a bit behind on the latest 😅 I noticed that in the ADR we mentioned that this field can be set with a time duration string. Would this still work if I provided something like "10h30m30s"?

Should we also add some type checks in case the casting causes a panic?

Reply:
Good question! But we don't have to do that since the Framework takes care of validating Vault's input fields. So that input would result in `rotation_window=37830` (seconds)

Reply:
I meant to say that the Framework handles validating the **types** of the input fields

Reply:
Ah nice, thanks for clarifying!